docs/api/v1/api.md

```diff
@@ -0,0 +1,1099 @@
+| GET | /projects/{project}/pipelines/{pipeline} |     | Pipeline | Get a pipeline | WIP, [link](#get-a-pipeline) |
+| DELETE | /projects/{project}/pipelines/{pipeline} |     |     | Delete a pipeline | WIP, [link](#delete-a-pipeline) |
+| POST | /projects/{project}/pipelines/{pipeline} | version:string</br>description:string</br>createTag:bool</br>stages:[]string</br> |     | Trigger a pipeline | WIP, [link](#trigger-a-pipeline) |
+| POST | /projects/{project}/pipelines/{pipeline}/gitLabWebhook |     |     | Trigger a pipeline by gitlab webhook | trigger by gitlab |
+| POST | /projects/{project}/pipelines/{pipeline}/gitHubWebhook |     |     | Trigger a pipeline by github webhook | trigger by github |
+
+### Pipeline Record APIs
+
+- [Pipeline Record Data Structure](#PipelineRecordDataStructure)
+
+| Method | SubPath | Request Body | Response Body | function | Detail |
+| --- | --- | --- | --- | --- | --- |
+| GET | /projects/{project}/pipelines/{pipeline}/records |     | Pipeline record list | Get all pipeline records | WIP, [link](#get-all-pipeline-records) |
+| GET | /projects/{project}/pipelines/{pipeline}/records/{recordId} |     | Pipeline record | Get a pipeline record | WIP, [link](#get-a-pipeline-record) |
+| DELETE | /projects/{project}/pipelines/{pipeline}/records/{recordId} |     |     | Delete a pipeline record | WIP, [link](#delete-a-pipeline-record) |
+| GET | /projects/{project}/pipelines/{pipeline}/records/{recordId}/logs |     | string | Get log of a pipeline record | WIP, [link](#get-log-of-a-pipeline-record) |
+
+## API Common
+
+### Prefix
+
+Add `/api/v1/` ahead, complete path: `/api/v1/` + `SubPath`
+
+### Path Parameter Explaination
+
+In path parameter, both `{project}` and `{pipiline}` are `name`, and only `{recordId}` is `ID`
+
+### Client Error Response
+
+```
+400 Bad Request
+
+{
+    "message": "human readable",
+    "reason": "human readable",
+    "details": "error details"
+}
+```
+
+## API Details
+
+### ProjectDataStructure
+
+```json
+{
+  "CreatedTime": "2017-08-23T09:44:08.653Z", // created time of the project
+  "Description": "string",                   // name of the project, should be unique
+  "ID": "string",                            // id of the project
+  "Name": "string",                          // name of the project, should be unique
+  "Owner": "string",                         // owner of the project
+  "UpdatedTime": "2017-08-23T09:44:08.653Z"  // updated time of the project
+}
+```
+
+Project is responsible for managing a set of applications, in this system, mainly managing a number of pipelining corresponding to this set of applications.
+
+### Add a project
+
+create a new project.
+
+#### Request
+
+Http method: `POST`
+
+URL: `/api/v1/projects`
+
+Header:
+
+```
+X-Tenant: any
+```
+
+Body:
+
+```
+{
+    "Description": "",                   // string, optional
+    "Name": "",                          // string, required
+    "Owner": ""                          // string, required
```

Comment:
The first letter of the fields should be lowercase. There are same errors in other places.